### DISCUSSION TITLE
Embedding Publii into Existing Site

### DISCUSSION THREAD

**Discussion by stellarpower** [GetPublii/Publii#1541] [General]
Hi,

Brand new to Publii, an not a web developer really - but, I was wondering how easy/difficult it might be to embed Publii content into an existing website. Basically using it as a CMS, but without outputting a full standalone site, and instead, rendering content within the existing site using its CSS styles, headers and footers, etc.

This existing site has a lot of theming applied, and rather than transition the whole thing, I am wondering if I can use Publii just to add new content, upload this statically to a particular location, and then the existing site could fetch the posts from this location and show them in a carousel or whatever within the existing site, embed the content using the existing site's theme if I can get in essence the contents of the WYSIWYG editor as unstyled plain straightforward HTML; maybe the list of posts and meta-info as JSON.

Is this something that can be achieved easily? Or would it require a good amount of work?

Thanks

**Comment by candidexmedia** [comment]
My first instinct would be to say no, but there may be tech-savvier folks out there who might have some ideas.

If I were in your shoes, I would:

1. set up the blog as a subdomain (if you're keen on using Publii), 
2. use a service that actually does it like BlogHandy, Quickblog, Dropinblog, etc. OR
3. just rebuild the site from scratch.

What is the main site built on?

**Reply by stellarpower** [reply]
Thanks, I'll look into those, or maybe I can hack into Publii and export the content in some format I can embed before full HTML rendering. The blog does need to be on the main site. Unfortunately, it's built using WordPress.

**Reply by candidexmedia** [reply]
Sounds good, and please share your findings if you do find a hack.

Btw: did you know that Publii has a Wordpress import tool? Not sure if it includes CSS, but the tool is there.

**Reply by stellarpower** [reply]
Will do! And yes, I saw this thanks, would be very useful. I may see if some other static CMSs exist with WYSIWYG editors that can just output something more like JSON, but if not will try taking a dive in, and it's useful if I can bring in the existing blog posts.
